### PR TITLE
No carbon tax break net profits

### PR DESCRIPTION
**Pull request by frmfisc**

### CODE REVIEW

File: R/financial_functions.R [hunk 1]
```diff
@@ -45,23 +45,75 @@ calculate_net_profits <- function(data,
     )
 }
 
+#' Calculates annual net profits on the company-technology level for the
+#' baseline and late and sudden scenarios without carbon tax Climate laggards which need to build
+#' out their production in increasing technologies to compensate for their
```
Comment (MirijaHa) [inline R/financial_functions.R:50]: 1. please use the roxygen sceletons to add documentation to make sure that they are formatted corretly and complete. Simply click on a function and click Insert Roxygen Sceleton under the Code pain
2. I would suggest to put the details not in calculate_net_profits_without_carbon_tax (here it is enought to say that it calculates the net profits for the following cases) and instead go into detail on the smaller functions.

File: R/financial_functions.R [hunk 2]
```diff
@@ -45,23 +45,75 @@ calculate_net_profits <- function(data,
...
+#'
+#' @param data A data frame containing the production forecasts of companies
+#'   under baseline and late and sudden, market prices/costs, company net profit
+#'   margins, the proximity to target in the production forecast period and an
+#'   indication of the direction of the technology.
+#'   @param direction either declining or increasing
+
 calculate_net_profits_without_carbon_tax <- function(data) {
+  baseline <- calculate_net_profits_baseline(data)
+  shock_increasing_technologies <- calculate_net_profits_shock_increasing_technologies(data = data %>% dplyr::filter(direction == "increasing"))
```
Comment (MirijaHa) [inline R/financial_functions.R:70]: did you run styler already?

File: R/financial_functions.R [hunk 3]
```diff
@@ -45,23 +45,75 @@ calculate_net_profits <- function(data,
...
+
+#' Calculates annual net profits on the company-technology level for the
+#' baseline
+#' @param data A data frame containing the production forecasts of companies
+#'   under baseline and late and sudden, market prices/costs, company net profit
+#'   margins, the proximity to target in the production forecast period and an
+#'   indication of the direction of the technology.
+
+calculate_net_profits_baseline <- function(data) {
+  data <- data %>%
```
Comment (MirijaHa) [inline R/financial_functions.R:85]: I would suggest to explicitly return from functions, i.e. put return(data) at the end. It is not always needed, but there is some rare cases where not returning explicitly will return the wrong thing..

File: R/financial_functions.R [hunk 4]
```diff
@@ -45,23 +45,75 @@ calculate_net_profits <- function(data,
...
+#'   indication of the direction of the technology.
+#'   @param direction either declining or increasing
+
 calculate_net_profits_without_carbon_tax <- function(data) {
+  baseline <- calculate_net_profits_baseline(data)
+  shock_increasing_technologies <- calculate_net_profits_shock_increasing_technologies(data = data %>% dplyr::filter(direction == "increasing"))
+  shock_declining_technologies <- calculate_net_profits_shock_declining_technologies(data = data %>% dplyr::filter(direction == "declining"))
+
+  data_test <- dplyr::full_join(shock_increasing_technologies, shock_declining_technologies)
+  data_test <- dplyr::full_join(data_test, baseline)
```
Comment (MirijaHa) [inline R/financial_functions.R:74]: data_test is not the nicest name

**Review by MirijaHa — COMMENTED** [review]
Mostly just formalities. I think that the code is already much easier to understand thanks to your refacotoring 🚀 

This would be a good point in time to add unit tests for the functions 🤓

**Review by MirijaHa — COMMENTED** [review]
please also fix the new note (no global binding for direction)

File: R/financial_functions.R (hunk 3 above)
Comment (frmfisc) [reply R/financial_functions.R:85]: done

File: R/financial_functions.R (hunk 4 above)
Comment (frmfisc) [reply R/financial_functions.R:74]: changed it back to data

File: R/financial_functions.R (hunk 1 above)
Comment (frmfisc) [reply R/financial_functions.R:50]: 2. makes sense, got it. 1. still needs to be addressed

File: R/financial_functions.R (hunk 2 above)
Comment (frmfisc) [reply R/financial_functions.R]: actually yes

File: R/financial_functions.R (hunk 1 above)
Comment (frmfisc) [reply R/financial_functions.R]: 1. done

**Pull request merged**